ci/minimum_supported_agent_version_test.go

```diff
@@ -0,0 +1,104 @@
+//go:build ci
+
+package ci
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"path"
+	"strings"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/ops-agent/integration_test/metadata"
+	"gopkg.in/yaml.v2"
+)
+
+var (
+	added    = flag.String("added", "", "Added files")
+	modified = flag.String("modified", "", "Modified files")
+	labels   = flag.String("labels", "", "Pull request labels")
+	baseDir  = flag.String("base-dir", "", "Directory path for the base of the pull request")
+)
+var basePath = os.Getenv("GITHUB_WORKSPACE")
+
+func TestMinimumSupportedAgentVersionIsModified(t *testing.T) {
+	if basePath == "" {
+		t.Fatal("Must be run within git action")
+	}
+
+	if strings.Contains(*labels, "release") {
+		return
+	}
+
+	addedMetadataFilePaths := filterForMetadata(*added)
+	modifiedMetadataFilePaths := filterForMetadata(*modified)
+
+	if len(addedMetadataFilePaths) != 0 {
+		testAddedMetadata(t, addedMetadataFilePaths)
+	}
+
+	if len(modifiedMetadataFilePaths) != 0 {
+		testModifiedMetadata(t, modifiedMetadataFilePaths)
+	}
+}
+
+func filterForMetadata(filePaths string) []string {
+	metadataFilePaths := make([]string, 0)
+	for _, filePath := range strings.Split(filePaths, " ") {
+		if strings.Index(filePath, "metadata.yaml") > -1 {
+			metadataFilePaths = append(metadataFilePaths, filePath)
+		}
+	}
+	return metadataFilePaths
+}
+
+func testAddedMetadata(t *testing.T, dirs []string) {
+	for _, dir := range dirs {
+		dir := dir
+		t.Run(dir, func(t *testing.T) {
+			t.Parallel()
+			m := metadata.IntegrationMetadata{}
+			if err := unmarshalYamlFromFile(path.Join(basePath, dir), &m); err != nil {
+				t.Fatal(err)
+			}
+
+			var emptyMetadata metadata.MinimumSupportedAgentVersion
+			if m.MinimumSupportedAgentVersion != emptyMetadata {
+				t.Fatal("Cannot add metadata.MinimumSupportedAgentVersion")
+			}
```

Comment:
Trying to understand those lines. It could be my problem..

So `var emptyMetadata metadata.MinimumSupportedAgentVersion` sets a by default null or empty `MinimumSupportedAgentVersion`
Then the next if statement compares if the parsed added file has the same `MinimumSupportedAgentVersion` with the empty one?  I guess you were trying to check if the parsed one is not empty then good. If it's empty and meaningless, it shall fail the test?

Reply:
I think i get it now. If the PR label contains release, then always return early and true. If not, and it contains changes of `MinimumSupportedAgentVersion` in either added or modified files, fail it. Though i am still not sure if a developer put below in the metadata.yaml file, does that count as emptyMetadata file and the PR will pass the test which it should not
```
...
minimum_supported_agent_version:
....
```

Reply:
Changed in: [95aa45e](https://github.com/GoogleCloudPlatform/ops-agent/pull/775/commits/95aa45e4655aa3a43fc8943102afaf47950eb7c7)